how-to/containerd-kata.md

```diff
@@ -0,0 +1,357 @@
+  - [Launch Pods with crictl command line](#launch-pods-with-crictl-command-line)
+
+This document covers the installation and configuration of [containerd](https://containerd.io/) 
+and [Kata Containers](https://katacontainers.io). The containerd will provide not only the `ctr`
+command line tool, but also the [CRI](https://kubernetes.io/blog/2016/12/container-runtime-interface-cri-in-kubernetes/) 
+interface for [Kubernetes](https://kubernetes.io) and other CRI clients.
+
+This document is mainly written for Kata Containers v1.5.0-rc2 or above and containerd v1.2.0 or above.
+The document addresses the previous versions as well though we suggest users upgrade to the newer versions for better support.
+
+## Concepts
+
+### Kubernetes RuntimeClass
+
+[RuntimeClass](https://kubernetes.io/docs/concepts/containers/runtime-class/) is a Kubernetes feature first 
+introduced in Kubernetes 1.12 as alpha. It is the feature for selecting the container runtime configuration to 
+use to run a pod’s containers. And this feature is supported in `containerd` since [v1.2.0](https://github.com/containerd/containerd/releases/tag/v1.2.0).
+
+Before the RuntimeClass was introduced, Kubernetes wasn't aware of the difference of runtimes on the node. `kubelet`
+creates Pod sandboxes and containers through CRI implementations and treat all the Pods equally. However, there
+are requirements to run trusted Pod (such as kubernetes plugin) in native container like runC and run untrusted
+workload with isolated sandboxes such as Kata Containers.
+
+As a result, the CRI implementations extended their semanitics for the requirements:
+
+- At the beginning, [frakti](https://github.com/kubernetes/frakti) check the network configuration of a Pod, and
+  treat Pod with `host` network as trusted, while others as untrusted;
+- The containerd introduced an annotation for untrusted Pod since [v1.0](https://github.com/containerd/cri/blob/v1.0.0-rc.0/docs/config.md):
+  ```yaml
+  annotations:
+     io.kubernetes.cri.untrusted-workload: "true"
+  ```
+- Similarly, CRI-O introduced the annotation `io.kubernetes.cri-o.TrustedSandbox` for the same case.
+
+To eliminate the complexity of user configuration introduced by the non-standardized annotations and provide 
+extensibility, `RuntimeClass` was introduced and users could affect the runtime behavior through it without 
+the knowledge of the CRI daemons. We suggest the users have multiple runtimes use `RuntimeClass` instead of 
+the deprecated annotations.
+
+### Containerd Runtime V2 API (Shim V2 API)
+
+The [`containerd-shim-kata-v2` (short as `shimv2` in this documentation)](https://github.com/kata-containers/runtime/tree/master/containerd-shim-v2) 
+implements the [Containerd Runtime V2 (Shim API)](https://github.com/containerd/containerd/tree/master/runtime/v2) for Kata.
+With `shimv2`, kubernetes could launch Pod and OCI compatible containers with one shim (the `shimv2`) per Pod instead
```

Comment:
Lines 59-61 suggested rewrite - although I might have changed meaning because I am unclear on what the standalone kata-proxy piece refers to: 
With `shimv2`, kubernetes can launch Pod and OCI-compatible containers with one shim per Pod. Prior to `shimv2`, `2N+1` shims (i.e. a `containerd-shim` and a `kata-shim` for each container and the Pod sandbox itself) and no standalone `kata-proxy` process were used, even with vsock not available.